xxhash.h

```diff
@@ -5146,8 +5146,15 @@ XXH3_hashLong_64b_internal(const void* XXH_RESTRICT input, size_t len,
  * It's important for performance to transmit secret's size (when it's static)
  * so that the compiler can properly optimize the vectorized loop.
  * This makes a big performance difference for "medium" keys (<1 KB) when using AVX instruction set.
+ * When the secret size is unknown, or on GCC 12 where the mix of NO_INLINE and FORCE_INLINE
```

Comment:
The comment explains that, when secret's size is `static`,
then it's advantageous for performance to pass that to the function.
This suggests that the constant size is propagated inside `XXH3_hashLong_64b_withSecret()` for further optimization.

Of course, if we use `XXH_NO_INLINE`, this property is lost.

Now, this could be nonetheless an acceptable trade-off if the goal is to survive the specific `gcc-12 -Og` compilation scenario.

But my understanding of below macro test is that it's more general : 
1) It's applicable to any current and future version of `gcc`, >= 12
2) `XXH_NO_INLINE` is _always_ set (not just when `-Og` is set), with the exception when inlining is expressly required with `XXH_INLINE_ALL`.

So, effectively, this will impact performance on a broader range of scenarios.

And I wonder if this is the right trade off.

Reply:
When XXH3 is not inlined, there isn't much of a benefit to inlining XXH3_hashLong_withSecret, as the secret size would be entirely unknown.

Reply:
That's a good point.
For some reason, I expected this variant to be within the call chain of the "default" variant using the default `kSecret`,
but it's not, "default" delegates directly to `XXH3_hashLong_64b_internal()`.